src/meshes/transfinite_mappings_3d.jl

```diff
@@ -0,0 +1,172 @@
+
+  coordinate = zeros(eltype(xi), 3)
+  for j in 1:3
+    coordinate[j] += (0.125 * ( corner_points[j, 1] * (1.0 - xi) * (1.0 - eta) * (1.0 - zeta)
+                              + corner_points[j, 2] * (1.0 + xi) * (1.0 - eta) * (1.0 - zeta)
+                              + corner_points[j, 3] * (1.0 + xi) * (1.0 + eta) * (1.0 - zeta)
+                              + corner_points[j, 4] * (1.0 - xi) * (1.0 + eta) * (1.0 - zeta)
+                              + corner_points[j, 5] * (1.0 - xi) * (1.0 - eta) * (1.0 + zeta)
+                              + corner_points[j, 6] * (1.0 + xi) * (1.0 - eta) * (1.0 + zeta)
+                              + corner_points[j, 7] * (1.0 + xi) * (1.0 + eta) * (1.0 + zeta)
+                              + corner_points[j, 8] * (1.0 - xi) * (1.0 + eta) * (1.0 + zeta) ) )
+  end
+
+  return coordinate
+end
+
+
+# Construct the (x, y, z) node coordinates in the volume of a straight sided hexahedral element
+function calc_node_coordinates!(node_coordinates::AbstractArray{<:Any, 5}, element, nodes, corners)
+
+  for k in eachindex(nodes), j in eachindex(nodes), i in eachindex(nodes)
+    node_coordinates[:, i, j, k, element] .= straight_side_hex_map(nodes[i], nodes[j], nodes[k], corners)
+  end
+
+  return node_coordinates
+end
+
+
+# Transfinite mapping formula from a point (xi, eta, zeta) in reference space [-1,1]^3 to a point
+# (x,y,z) in physical coordinate space for a hexahedral element with general curved sides
+# See Section 4.3
+# - Andrew R. Winters (2014)
+#   Discontinous Galerkin spectral element approximations for the reflection and
+#   transmission of waves from moving material interfaces
+#   [PhD thesis, Florida State University](https://diginole.lib.fsu.edu/islandora/object/fsu%3A185342)
+function transfinite_hex_map(xi, eta, zeta, face_curves::AbstractVector{<:CurvedFace})
+
+  coordinate = zeros(eltype(xi), 3)
+  face_values = zeros(eltype(xi), (3, 6))
+  edge_values = zeros(eltype(xi), (3, 12))
+  corners = zeros(eltype(xi), (3, 8))
+
+  # Compute values along the face edges
+  edge_values[:, 1] .= evaluate_at(SVector(xi  , -1.0), face_curves[1])
+  edge_values[:, 2] .= evaluate_at(SVector(1.0 , zeta), face_curves[1])
+  edge_values[:, 3] .= evaluate_at(SVector(xi  ,  1.0), face_curves[1])
+  edge_values[:, 4] .= evaluate_at(SVector(-1.0, zeta), face_curves[1])
+
+  edge_values[:, 5] .= evaluate_at(SVector(xi  , -1.0), face_curves[2])
+  edge_values[:, 6] .= evaluate_at(SVector(1.0 , zeta), face_curves[2])
+  edge_values[:, 7] .= evaluate_at(SVector(xi  ,  1.0), face_curves[2])
+  edge_values[:, 8] .= evaluate_at(SVector(-1.0, zeta), face_curves[2])
+
+  edge_values[:, 9]  .= evaluate_at(SVector(eta, -1.0), face_curves[6])
+  edge_values[:, 10] .= evaluate_at(SVector(eta, -1.0), face_curves[4])
+  edge_values[:, 11] .= evaluate_at(SVector(eta,  1.0), face_curves[4])
+  edge_values[:, 12] .= evaluate_at(SVector(eta,  1.0), face_curves[6])
```

Comment:
```suggestion
  edge_values[:, 1] .= evaluate_at(SVector(xi  , -1), face_curves[1])
  edge_values[:, 2] .= evaluate_at(SVector(1 , zeta), face_curves[1])
  edge_values[:, 3] .= evaluate_at(SVector(xi  ,  1), face_curves[1])
  edge_values[:, 4] .= evaluate_at(SVector(-1, zeta), face_curves[1])

  edge_values[:, 5] .= evaluate_at(SVector(xi  , -1), face_curves[2])
  edge_values[:, 6] .= evaluate_at(SVector(1 , zeta), face_curves[2])
  edge_values[:, 7] .= evaluate_at(SVector(xi  ,  1), face_curves[2])
  edge_values[:, 8] .= evaluate_at(SVector(-1, zeta), face_curves[2])

  edge_values[:, 9]  .= evaluate_at(SVector(eta, -1), face_curves[6])
  edge_values[:, 10] .= evaluate_at(SVector(eta, -1), face_curves[4])
  edge_values[:, 11] .= evaluate_at(SVector(eta,  1), face_curves[4])
  edge_values[:, 12] .= evaluate_at(SVector(eta,  1), face_curves[6])
```
`1` will be promoted correctly and is less leaky than `1.0`, e.g.
```julia
julia> using StaticArrays

julia> x = 2.f0
2.0f0

julia> SVector(x, 1.0)
2-element SVector{2, Float64} with indices SOneTo(2):
 2.0
 1.0

julia> SVector(x, 1)
2-element SVector{2, Float32} with indices SOneTo(2):
 2.0
 1.0
```